rosflight/include/rosflight/mavrosflight/logger_interface.h

```diff
@@ -0,0 +1,202 @@
+/*
+ * Software License Agreement (BSD-3 License)
+ *
+ * Copyright (c) 2020 Jacob Willis.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file logger_interface.h
+ * @author Jacob Willis <jbwillis272@gmail.com>
+ */
+
+#ifndef MAVROSFLIGHT_LOGGER_INTERFACE_H
+#define MAVROSFLIGHT_LOGGER_INTERFACE_H
+
+#include <sstream>
+
+#include <stdio.h>
+
+namespace mavrosflight
+{
+/**
+ * \class LoggerInterface
+ * \brief Abstract base class for message handler
+ *
+ * The implementations of this class define how messages are displayed, logged,
+ * etc. To create custom behavior, a derived class should implement each of the
+ * public functions.
+ */
+template <typename Derived>
+class LoggerInterface
+{
+public:
+  template <typename... T>
```

Comment:
I don't have strong feelings, but maybe `template <typename... Args>`? I think that makes it a bit clearer at a glance that the template parameters are for the variadic arguments